backend/uclapi/workspaces/occupeye/archive.py

```diff
@@ -0,0 +1,219 @@
+                logging.info(f"      [+] Updating {validate['hardware_id']} -> {sensor['HardwareID']}")
+                logging.info(f"      [+] Updating {validate['survey_device_id']} -> {sensor['SurveyDeviceID']}")
+                obj = Sensors.objects.get(sensor_id=sensor["SensorID"])
+
+                time = datetime.strptime(sensor["TriggerDate"], "%Y-%m-%d")
+                time = time.replace(tzinfo=timezone.utc)
+                SensorReplacements.objects.create(sensor_id=sensor["SensorID"],
+                                                  survey_id=validate["survey_id"],
+                                                  hardware_id=validate['hardware_id'],
+                                                  survey_device_id=validate['survey_device_id'],
+                                                  datetime=time)
+
+                obj.hardware_id = sensor["HardwareID"]
+                obj.survey_device_id = sensor["SurveyDeviceID"]
+                obj.save()
+                self.current_sensors[str(sensor["SensorID"]) + "-" + str(survey.survey_id)] = {
+                    "hardware_id": sensor["HardwareID"], "survey_device_id": sensor["SurveyDeviceID"],
+                    "survey_id": survey.survey_id, "last_value": -1}
+            else:
+                logging.info(f"      [+] Sensor in different location "
+                             f"{validate['survey_id']} -> {survey.survey_id}")
+                Sensors.objects.create(sensor_id=sensor["SensorID"], hardware_id=sensor["HardwareID"],
+                                       survey_device_id=sensor["SurveyDeviceID"],
+                                       survey_id=survey.survey_id)
+                self.current_sensors[str(sensor["SensorID"]) + "-" + str(survey.survey_id)] = {
+                    "hardware_id": sensor["HardwareID"], "survey_device_id": sensor["SurveyDeviceID"],
+                    "survey_id": survey.survey_id, "last_value": -1}
+
+            logging.info(f"   [+] Continuing {start_date} to {end_date} ", end="")
+
+    def update(self):
+        self.current_sensors = self.get_current_sensors()
+
+        end_datetime = datetime.now().replace(microsecond=0, second=0, minute=0, hour=0)
+        for survey in Surveys.objects.all():
+            logging.info(f"[+] Updating {survey.survey_id} from {survey.last_updated} to {end_datetime}")
+            for step in self.time_steps(survey.last_updated, end_datetime):
+                start_date = step[0].strftime("%Y-%m-%d")
+                end_date = step[1].strftime("%Y-%m-%d")
+
+                logging.info(f"   [+] Requesting {start_date} to {end_date} ", end="")
+                updates = 0
+                sensors = self._endpoint.request(
+                    self._const.URL_ARCHIVE.format(start_date, end_date, survey.survey_id))
+
+                # Sensors where the hardware id or survey device id have changed will have two sets of data
```

Comment:
```suggestion
                # Sensors where the hardware id or survey device id have changed will have two sets of identical data
```

Reply:
The data is not always identical, I think the point at which they remove the sensor it changes.